signature-aggregator/config/source_blockchain.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	baseCfg "github.com/ava-labs/awm-relayer/config"
+	"github.com/ava-labs/awm-relayer/utils"
+)
+
+type SourceBlockchain struct {
+	SubnetID     string            `mapstructure:"subnet-id" json:"subnet-id"`
+	BlockchainID string            `mapstructure:"blockchain-id" json:"blockchain-id"` //nolint:lll
+	VM           string            `mapstructure:"vm" json:"vm"`
+	RPCEndpoint  baseCfg.APIConfig `mapstructure:"rpc-endpoint" json:"rpc-endpoint"` //nolint:lll
+	WSEndpoint   baseCfg.APIConfig `mapstructure:"ws-endpoint" json:"ws-endpoint"`   //nolint:lll
+
+	// convenience fields to access parsed data after initialization
+	subnetID     ids.ID
+	blockchainID ids.ID
+}
```

Comment:
Once ACP-118 is in place, will we need to configure `SourceBlockchains` at all?

I think we can hopefully get away without it entirely. Each request will specify the `sourceChainID` (used in the `AppRequest` [here](https://github.com/ava-labs/awm-relayer/blob/signature-aggregation-api/lib/aggregator.go#L140)), and the `signingSubnetID`, which determines which nodes the request should be sent to [here](https://github.com/ava-labs/awm-relayer/blob/signature-aggregation-api/lib/aggregator.go#L92). That should be sufficient for sending the proper request to the correct set of nodes, no matter the source blockchain.

If we're able to go this direction, I'd recommend that we start now by assuming all source subnets use subnet-evm's codec for signature requests. It won't work for the C-Chain in the immediate interim period, but then we can simply drop in the ACP-118 proto in place of the subnet-evm code for formatting the `AppRequest`s.

Reply:
Great point! Thank you. I assume that to do this we would need to be able to map a `sourceChainID` to `signingSubnetID` somehow. If not we can just make `signingSubnetID` a required field in the interface?

Reply:
> I assume that to do this we would need to be able to map a sourceChainID to signingSubnetID somehow. If not we can just make signingSubnetID a required field in the interface?

There's no mapping between `sourceChainID` and `signingSubnetID` (for instance, if the source chain is the P-chain, then the signing chain could be any subnet), but I think it'd be reasonable to either have `signingSubnetID` be a required param or to have the `signingSubnetID` default to the subnet containing the `sourceChainID` if it is not explicitly set.